### DISCUSSION TITLE
How do I add paramters to the footer of the PDF when rendering the quarto document of format typst

### DISCUSSION THREAD

**Discussion by Prav14** [quarto-dev/quarto-cli#8677] [Q&A]
### Description

Hi,
I would like to add the parameters I passed to the quarto document in the footer of the rendered document. I using typst to render the PDF. I tried using `r params$text_label` and also {{ < meta params.text_label > } } but unable to get the parameter value into the footer.

Code to render the quarto document
quarto::quarto_render("test_app_pdf1.qmd",execute_params=list(text_label="This is PDF generation test."))

quarto document code 
---
title: "Test Parametes in footer"
params: 
  text_label: " "
format: 
  typst:
    include-in-header: 
      text: |
        #set page(footer: [This is a footer; `r params$text_label`])
---
# This is a Quarto Test

I was wondering if there was another way to add the parameters to the footer of the PDF or if anyone could help to debug if I missed something in the above code.

I am using Quarto 1.4

Thank you!

**Comment by mcanouil** [comment]
Could you properly format your code to be usable? Thank you.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by Prav14** [comment]
Sure!
Code to render the quarto document
quarto::quarto_render("test_app_pdf1.qmd",execute_params=list(text_label="This is PDF generation test."))

Code of test_app_pdf1.qmd Quarto document
````qmd

---
title: "Test Parametes in footer"
params: 
  text_label: "In the qmd"
format: 
  typst:
    include-in-header: 
      text: |
        #set page(footer: [This is a footer; `r params$text_label`])
---

# This is a Quarto Test
````

Quarto check output

````qmd
```bash
Quarto 1.4.549
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.549
      Path: C:\Program Files\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.2.2
      Path: C:/PROGRA~1/R/R-42~1.2
      LibPaths:
        - C:/Users/MULUKPX/AppData/Local/R/win-library/4.2
        - C:/Program Files/R/R-4.2.2/library
      knitr: 1.42
      rmarkdown: 2.20

[>] Checking Knitr engine render......OK

```
````

**Reply by mcanouil** [reply]
Thanks.
I meant to edit your original post mostly.

You cannot do what you are trying to do.
Inline code in YAML is kind of a legacy of `knitr`.

Here is a Quarto workaround/approach:

````qmd
---
title: "Test Parametes in footer"
params: 
  text_label: "In the qmd"
format:
  typst:
    include-in-header: 
      text: |
        #set page(footer: [This is a footer; {{< meta params.text_label >}}])
---

# This is a Quarto Test
````

**Reply by Prav14** [reply]
Thanks for the reply. I actually tried this. But the problem I see when I use this is - it prints the parameter value from the quarto document (In the qmd) not the parameter value I pass when I render the quarto document (This is PDF generation test).

Below is the code I use to render the document.
quarto::quarto_render("test_app_pdf1.qmd",execute_params=list(text_label="This is PDF generation test."))

**Reply by mcanouil** [reply]
I don't think you can do what you want.

You can use `metadata` instead (it's available to the R wrapper package, see <https://quarto-dev.github.io/quarto-r/reference/quarto_render.html>).

```sh
quarto render index.qmd --metadata text_label:"oh no"
```

````qmd
---
title: "Test Parametes in footer"
text_label: "In the qmd"
format:
  typst:
    include-in-header: 
      text: |
        #set page(footer: [This is a footer; {{< meta text_label >}}])
---

# This is a Quarto Test
````
